src/rdkafka_broker.c

```diff
@@ -4447,14 +4446,14 @@ rd_kafka_fetch_reply_handle (rd_kafka_broker_t *rkb,
 				case RD_KAFKA_RESP_ERR_MSG_SIZE_TOO_LARGE:
 				default: /* and all other errors */
 					rd_dassert(tver->version > 0);
-					rd_kafka_q_op_err(
-						rktp->rktp_fetchq,
-						RD_KAFKA_OP_CONSUMER_ERR,
-						hdr.ErrorCode, tver->version,
-						rktp,
-						rktp->rktp_offsets.fetch_offset,
-						"Fetch failed: %s",
-						rd_kafka_err2str(hdr.ErrorCode));
+                                        rd_kafka_consumer_err(
```

Comment:
unconvinced this is the right place to be sending notifications of errors we're not expecting here. why not the standard error path? if we send them there, the user will probably have default logging set up which is an appropriate action. also, that's in line with what our story is for what error notifications are about. this muddles things up. i understand you are worried about maintaining ordering, but in what way does that help here?

Reply:
Apart from the Replica fix this PR simply refactors the consumer error code but does not change the current behaviour (except for one place),   so let's have the error-raising discussion outside the PR.